source/collections/VectorList.ooc

```diff
@@ -123,11 +123,10 @@ VectorList: class <T> {
 		for (i in 0 .. this count)
 			this[i] = function(this[i])
 	}
-	map: func <T, S> (function: Func(T) -> S) -> This<S> {
+	map: func <S> (function: Func(T) -> S) -> This<S> {
 		result := This<S> new(this count)
-		for (i in 0 .. this count) {
+		for (i in 0 .. this count)
 			result add(function(this[i]))
-		}
 		result
 	}
 	fold: func <T, S> (function: Func(T, S) -> S, initial: S) -> S {
```

Comment:
So it appears that we for now have to live with the fact that we must qualify ```S``` under certain circumstances, such as when ```S``` is used in an argument in a ```Func```, as is the case with ```VectorList fold()```.